domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Controllers/TransferAgreementAuditsControllerTests.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using API.ApiModels.Requests;
+using API.ApiModels.Responses;
+using API.Data;
+using API.IntegrationTests.Factories;
+using FluentAssertions;
+using Xunit;
+
+namespace API.IntegrationTests.Controllers;
+
+public class TransferAgreementAuditsControllerTests : IClassFixture<TransferAgreementsApiWebApplicationFactory>
+{
+    private readonly TransferAgreementsApiWebApplicationFactory factory;
+    private readonly HttpClient authenticatedClient;
+
+    public TransferAgreementAuditsControllerTests(TransferAgreementsApiWebApplicationFactory factory)
+    {
+        this.factory = factory;
+    }
+
+    [Fact]
+    public async Task Create_ShouldGenerateAudit_WhenTransferAgreementIsCreated()
+    {
+        var senderId = Guid.NewGuid();
+        var authenticatedClient = factory.CreateAuthenticatedClient(sub: senderId.ToString());
+
+        var transferAgreement = new CreateTransferAgreement(
+            DateTimeOffset.UtcNow.ToUnixTimeSeconds(),
+            DateTimeOffset.UtcNow.AddDays(1).ToUnixTimeSeconds(),
+            "12345678"
+        );
+
+        var createResponse = await authenticatedClient.PostAsJsonAsync("api/transfer-agreements", transferAgreement);
+        createResponse.EnsureSuccessStatusCode();
+
+        createResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var createdTransferAgreement = await createResponse.Content.ReadFromJsonAsync<TransferAgreementDto>();
+        createdTransferAgreement.Should().NotBeNull();
+
+        var auditsResponse = await authenticatedClient.GetAsync($"api/audits/transfer-agreements/{createdTransferAgreement.Id}");
```

Comment:
Consider using `authenticatedClient.GetFromJsonAsync<T>` if you just want to get the body of the response. If the API returns a status code different than "200 OK", an exception will be thrown.

Reply:
I understand what you mean.

I prefer to do things one step at a time. It makes it easier to tell a story, in the tests, and also makes it easier to debug.

Reply:
Changed to your suggestion